src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SelectedMembers/AbstractSelectedMembers.cs

```diff
@@ -28,13 +29,19 @@ internal abstract class AbstractSelectedMembers<
         where TVariableSyntax : SyntaxNode
     {
         protected abstract SyntaxList<TMemberDeclarationSyntax> GetMembers(TTypeDeclarationSyntax containingType);
-        protected abstract IEnumerable<TVariableSyntax> GetAllDeclarators(TFieldDeclarationSyntax field);
-
-        protected abstract SyntaxToken GetVariableIdentifier(TVariableSyntax declarator);
-        protected abstract SyntaxToken GetPropertyIdentifier(TPropertyDeclarationSyntax declarator);
+        protected abstract IEnumerable<(SyntaxToken identifier, SyntaxNode declaration)> GetDeclaratorsAndIdentifiers(TMemberDeclarationSyntax member);
```

Comment:
nit: method name is 'get declarator and identifier' but returns a tuple in reverse order.